diff --git a/src/notification.c b/src/notification.c
index f1cc14d..9d16d04 100644
--- a/src/notification.c
+++ b/src/notification.c
@@ -491,6 +491,11 @@ static void notification_format_message(struct notification *n)
         /* truncate overlong messages */
         if (strnlen(n->msg, DUNST_NOTIF_MAX_CHARS + 1) > DUNST_NOTIF_MAX_CHARS) {
                 char * buffer = g_strndup(n->msg, DUNST_NOTIF_MAX_CHARS);
+                buffer[DUNST_NOTIF_MAX_CHARS-4] = '.';
+                buffer[DUNST_NOTIF_MAX_CHARS-3] = '.';
+                buffer[DUNST_NOTIF_MAX_CHARS-2] = '.';
+                buffer[DUNST_NOTIF_MAX_CHARS-1] = '\0';
+
                 g_free(n->msg);
                 n->msg = buffer;
         }
@@ -559,14 +564,14 @@ void notification_update_text_to_render(struct notification *n)
                 char *new_buf;
                 if (hours > 0) {
                         new_buf =
-                            g_strdup_printf("%s (%ldh %ldm %lds old)", buf, hours,
+                            g_strdup_printf("%s\n%ldh %02ldm %02lds", buf, hours,
                                             minutes, seconds);
                 } else if (minutes > 0) {
                         new_buf =
-                            g_strdup_printf("%s (%ldm %lds old)", buf, minutes,
+                            g_strdup_printf("%s\n%ldm %02lds", buf, minutes,
                                             seconds);
                 } else {
-                        new_buf = g_strdup_printf("%s (%lds old)", buf, seconds);
+                        new_buf = g_strdup_printf("%s\n%lds", buf, seconds);
                 }
 
                 g_free(buf);
diff --git a/src/notification.h b/src/notification.h
index d78d5f9..e3a18e5 100644
--- a/src/notification.h
+++ b/src/notification.h
@@ -8,7 +8,7 @@
 
 #include "markup.h"
 
-#define DUNST_NOTIF_MAX_CHARS 50000
+#define DUNST_NOTIF_MAX_CHARS 250
 
 enum behavior_fullscreen {
         FS_NULL,      //!< Invalid value
